There are two issues now. 

1. It doesn't make sense Travis fails given that what I change is only one line of dockerfile which is not even used there. It fails with 
```
Traceback (most recent call last):
  File "kubernetes_namespace.py", line 69, in <module>
    test()
  File "kubernetes_namespace.py", line 24, in test
    deploy_(conn_1)
  File "kubernetes_namespace.py", line 48, in deploy_
    registry=CLIPPER_CONTAINER_REGISTRY)
  File "/home/travis/build/ucbrise/clipper/clipper_admin/clipper_admin/deployers/python.py", line 91, in create_endpoint
    batch_size, pkgs_to_install)
  File "/home/travis/build/ucbrise/clipper/clipper_admin/clipper_admin/deployers/python.py", line 222, in deploy_python_closure
    registry, num_replicas, batch_size, pkgs_to_install)
  File "/home/travis/build/ucbrise/clipper/clipper_admin/clipper_admin/clipper_admin.py", line 412, in build_and_deploy_model
    container_registry, pkgs_to_install)
  File "/home/travis/build/ucbrise/clipper/clipper_admin/clipper_admin/clipper_admin.py", line 531, in build_model
    fileobj=context_file, custom_context=True, tag=image)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/docker/models/images.py", line 287, in build
    raise BuildError(chunk['error'], result_stream)
docker.errors.BuildError: manifest for clippertesting/python36-closure-container:75c9be50fd not found
```

2. Integration tests are failing with 
python2
```python2
[integration_py2_mxnet] 19-04-23:22:39:32 INFO [test_utils.py:197] Illegal instruction (core dumped)
```

python3
```python3
docker.errors.NotFound: 404 Client Error: Not Found ("No such container: ba23ddc3349f215b1dc2dbbb4b2da70382092f4730b660932775ba0fb2774c6d"
```

--- a/integration-tests/clipper_metric_docker.py
+++ b/integration-tests/clipper_metric_docker.py
@@ -41,6 +41,7 @@ def get_metrics_config():
 def get_matched_query(metric_addr, metric_name):
     query = gen_match_query(metric_addr, metric_name)
     logger.info("Querying: {}".format(query))
+    print('test why docker query is broken: {}'.format(repr(query)))
     res = requests.get(query).json()
     logger.info(res)
     return res

--- a/integration-tests/deploy_mxnet_models.py
+++ b/integration-tests/deploy_mxnet_models.py
@@ -13,7 +13,7 @@ cur_dir = os.path.dirname(os.path.abspath(__file__))
 import mxnet as mx
 
 from test_utils import (create_docker_connection, BenchmarkException, headers,
-                        log_clipper_state, log_docker)
+                        log_clipper_state, log_docker, log_model)
 
 cur_dir = os.path.dirname(os.path.abspath(__file__))
 sys.path.insert(0, os.path.abspath("%s/../clipper_admin" % cur_dir))
@@ -160,6 +160,7 @@ if __name__ == "__main__":
         except BenchmarkException:
             logger.exception("BenchmarkException")
             log_docker(clipper_conn)
+            log_model(clipper_conn, cluster_name)
             log_clipper_state(clipper_conn)
             clipper_conn = create_docker_connection(
                 cleanup=True, start_clipper=False, cleanup_name=cluster_name)
@@ -170,6 +171,7 @@ if __name__ == "__main__":
     except Exception:
         logger.exception("Exception")
         log_docker(clipper_conn)
+        log_model(clipper_conn, cluster_name)
         log_clipper_state(clipper_conn)
         clipper_conn = create_docker_connection(
             cleanup=True, start_clipper=False, cleanup_name=cluster_name)

--- a/integration-tests/deploy_mxnet_models.py
+++ b/integration-tests/deploy_mxnet_models.py
@@ -13,7 +13,7 @@ cur_dir = os.path.dirname(os.path.abspath(__file__))
 import mxnet as mx
 
 from test_utils import (create_docker_connection, BenchmarkException, headers,
-                        log_clipper_state, log_docker, log_model)
+                        log_clipper_state, log_docker, log_cluster_model)
 
 cur_dir = os.path.dirname(os.path.abspath(__file__))
 sys.path.insert(0, os.path.abspath("%s/../clipper_admin" % cur_dir))
@@ -160,7 +160,7 @@ if __name__ == "__main__":
         except BenchmarkException:
             logger.exception("BenchmarkException")
             log_docker(clipper_conn)
-            log_model(clipper_conn, cluster_name)
+            log_cluster_model(clipper_conn, cluster_name)
             log_clipper_state(clipper_conn)
             clipper_conn = create_docker_connection(
                 cleanup=True, start_clipper=False, cleanup_name=cluster_name)
@@ -171,7 +171,7 @@ if __name__ == "__main__":
     except Exception:
         logger.exception("Exception")
         log_docker(clipper_conn)
-        log_model(clipper_conn, cluster_name)
+        log_cluster_model(clipper_conn, cluster_name)
         log_clipper_state(clipper_conn)
         clipper_conn = create_docker_connection(
             cleanup=True, start_clipper=False, cleanup_name=cluster_name)

--- a/integration-tests/test_utils.py
+++ b/integration-tests/test_utils.py
@@ -195,3 +195,28 @@ def log_docker(clipper_conn):
         logger.info('Name {}, Image {}, Status {}, Label {}'.format(
             cont.name, cont.image, cont.status, cont.labels))
         logger.info(cont.logs())
+
+
+def log_cluster_model(clipper_conn, cluster_name):
+    if clipper_conn is None:
+        return
+
+    """Retrieve status and log for last ten containers"""
+    container_runing = clipper_conn.cm.docker_client.containers.list(limit=30, all=True)
+    logger.info('================Cluster model logs====================')
+    logger.info('It includes broken models')
+    logger.info('----------------------')
+    logger.info('Model container status (including broken one)')
+    for cont in container_runing:
+        if cluster_name in cont.name:
+            logger.info('Name {}, Image {}, Status {}, Label {}'.format(
+                cont.name, cont.image, cont.status, cont.labels))
+
+    logger.info('----------------------')
+    logger.info('Printing out model logs')
+
+    for cont in container_runing:
+        if cluster_name in cont.name:
+            logger.info('Name {}, Image {}, Status {}, Label {}'.format(
+                cont.name, cont.image, cont.status, cont.labels))
+            logger.info(cont.logs())

